packages/ember-glimmer/lib/ember-metal-views/index.js

```diff
@@ -13,12 +15,41 @@ class DynamicScope {
   }
 }
 
+function maybeUpdate(lastTagValue, renderedRoots) {
+  if (CURRENT_TAG.validate(lastTagValue)) { return; }
+  for (let i = 0; i < renderedRoots.length; ++i) {
+    let view = renderedRoots[i];
+    view.renderer.rerender(view);
+  }
+}
+
+function registerView(view, renderedRoots, scheduler) {
+  if (!renderedRoots.length) {
+    run.backburner.on('begin', scheduler);
+  }
+  renderedRoots.push(view);
+}
+
+function deregisterView(view, renderedRoots, scheduler) {
+  let viewIndex = renderedRoots.indexOf(view);
+  if (~viewIndex) {
+    renderedRoots.splice(viewIndex, 1);
+    if (!renderedRoots.length) {
+      run.backburner.off('begin', scheduler);
+    }
+  }
+}
+
 class Renderer {
   constructor({ dom, env, destinedForDOM = false }) {
     this._root = null;
     this._dom = dom;
     this._env = env;
     this._destinedForDOM = destinedForDOM;
+    this._renderedRoots = [];
+    this._scheduler = function scheduleMaybeUpdate() {
+      run.backburner.schedule('render', null, maybeUpdate, CURRENT_TAG.value(), this._renderedRoots);
+    }.bind(this);
```

Comment:
@stefanpenner I don't understand how this helps with the fact that the callback's context is lost once it is passed to `backburner.on`. Could you explain?